src/geff/interops/trackmate_xml.py

```diff
@@ -0,0 +1,583 @@
+
+    Returns:
+        nx.Graph: A NetworkX graph representing the TrackMate data.
+        dict[str, str]: A dictionary containing the units of the model, with keys
+            'spatialunits' and 'timeunits'.
+    """
+    graph = nx.Graph()
+
+    # So as not to load the entire XML file into memory at once, we're
+    # using an iterator to browse over the tags one by one.
+    # The events 'start' and 'end' correspond respectively to the opening
+    # and the closing of the considered tag.
+    it = ET.iterparse(xml_path, events=["start", "end"])
+    _, root = next(it)  # Saving the root of the tree for later cleaning.
+
+    for event, element in it:
+        if element.tag == "Model" and event == "start":
+            units = _get_units(element)
+            root.clear()  # Cleaning the tree to free up some memory.
+            # All the browsed subelements of `root` are deleted.
+
+        # Get the spot, edge and track features and add them to the
+        # attributes metadata.
+        if element.tag == "FeatureDeclarations" and event == "start":
+            attrs_md = _get_attributes_metadata(it, element)
+            root.clear()
+
+        # Adding the spots as nodes.
+        if element.tag == "AllSpots" and event == "start":
+            # TODO: segmentation will be used when GEFF supports polygons.
+            # segmentation = _add_all_nodes(it, element, attrs_md, graph)
+            _add_all_nodes(it, element, attrs_md, graph)
+            root.clear()
+
+        # Adding the tracks as edges.
+        if element.tag == "AllTracks" and event == "start":
+            # TODO: implement storage of track attributes.
+            # tracks_attrs = _build_tracks(it, element, attrs_md, graph)
+            _build_tracks(it, element, attrs_md, graph)
+            root.clear()
+
+            # Removal of filtered spots / nodes.
+            if discard_filtered_spots:
+                # Those nodes belong to no tracks: they have a degree of 0.
+                lone_nodes = [n for n, d in graph.degree if d == 0]
+                graph.remove_nodes_from(lone_nodes)
+
+        # Filtering out tracks.
+        if element.tag == "FilteredTracks" and event == "start":
+            # Removal of filtered tracks.
+            id_to_keep = _get_filtered_tracks_ID(it, element)
+            if discard_filtered_tracks:
+                to_remove = [n for n, t in graph.nodes(data="TRACK_ID") if t not in id_to_keep]
+                graph.remove_nodes_from(to_remove)
+
+        if element.tag == "Model" and event == "end":
+            break  # We are not interested in the following data.
+
+    return graph, units
+
+
+def from_trackmate_xml_to_geff(
+    xml_path: Path | str,
+    geff_path: Path | str,
+    discard_filtered_spots: bool = False,
+    discard_filtered_tracks: bool = False,
+    overwrite: bool = False,
+    zarr_format: Literal[2, 3] = 2,
+) -> None:
+    """
+    Convert a TrackMate XML file to a GEFF file.
```

Comment:
Your comments and documentation through this PR are amazing!

I'm just wondering if there is more trackmate specific information that might be needed here with regards to how different special properties are handled. I can't really judge myself because I'm not familiar enough with trackmate xml